library/core/src/str/pattern.rs

```diff
@@ -1965,3 +1978,26 @@ unsafe fn small_slice_eq(x: &[u8], y: &[u8]) -> bool {
         vx == vy
     }
 }
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+pub mod verify {
+    use super::*;
+
+    #[cfg(all(kani, target_arch = "x86_64"))] // only called on x86
+    #[kani::proof]
+    #[kani::unwind(4)]
+    pub fn check_small_slice_eq() {
```

Comment:
please add harnesses for empty slices with dangling pointer.

Reply:
I added one such test. But it fails and gives the same error even without loop contracts. I suppose another safety precondition of `small_slice_eq` should be not dangling pointer.